controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +200,81 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context, state *consumerReconcileState) {
+	r.keyIndex.mu.Lock()
+	defer r.keyIndex.mu.Unlock()
+
+	if state == nil || state.namespaceToConsumers == nil {
+		return
+	}
+
+	// Clear old indexes
+	r.keyIndex.index = make(map[string]map[string]map[string]string)
+
+	// Check conflicts for all valid consumers in the current state
+	for ns, consumers := range state.namespaceToConsumers {
+		if consumers == nil {
+			continue
+		}
+
+		validConsumers := make(map[string]*mosniov1.Consumer)
+
+		for name, consumer := range consumers {
+			if consumer == nil {
+				continue
+			}
+			// Create a new map to filter out invalid consumers
+			if err := r.indexConsumer(ns, consumer); err == nil {
+				validConsumers[name] = consumer
+			} else {
+				consumer.SetAccepted(mosniov1.ReasonInvalid, err.Error())
+			}
+		}
+		state.namespaceToConsumers[ns] = validConsumers
```

Comment:
According to the official Go documentation, it is safe to delete key-value pairs during iteration, but it is not safe to add new keys during iteration.